Repository/PhpcrRepository.php

```diff
@@ -119,4 +125,85 @@ protected function buildCollection(array $nodes)
 
         return $collection;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function add($path, $resource)
+    {
+        Assert::notEq('', trim($path, '/'), 'The root directory cannot be created.');
+        Assert::startsWith($path, '/', 'The target path %s is not absolute.');
+
+        $resolvedPath = $this->resolvePath($path);
+        try {
+            $parentNode = $this->session->getNode($resolvedPath);
+        } catch (PathNotFoundException $e) {
+            throw new InvalidArgumentException('No parent node created for '.$path, $e->getCode(), $e);
+        }
+
+        /** @var PhpcrResource[] $resources */
+        $resources = $resource instanceof IteratorAggregate ? $resource : new ArrayResourceCollection([$resource]);
+        Assert::isInstanceOf($resources, ResourceCollection::class, 'The list should be of instance "ResourceCollection".');
+
+        foreach ($resources as $resource) {
+            Assert::isInstanceOf($resource, CmfResource::class, 'The resource needs to of instance "CmfResource".');
```

Comment:
`needs to be an instance`. But, can we not use the default assertion message here? It would be more robust.